### PR TITLE
[bitnami/contour] Move envoy customReadinessProbe template to envoy container

### PR DESCRIPTION
**Pull request by seth-epps**
### Description of the change

The `envoy.customReadinessProbe` chart values are presently being applied to the shutdown manager and not the envoy container, meaning the shutdown manager probes are overridden by the defined probes.

I also noticed the daemonset container spec was missing this entirely so I added it there as well.

### Benefits

Fixes a bug in https://github.com/bitnami/charts/issues/24859

### Possible drawbacks

If someone relies on using the bug to define shutdown manager readiness checks, this will break it.

### Applicable issues

- fixes # https://github.com/bitnami/charts/issues/24859

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by seth-epps** [comment]
I noticed the order of the custom probe template values seem to precede the default block, but naively I think the custom block should override any enabled defaults. I can move it above if that makes more sense

File: bitnami/contour/templates/envoy/deployment.yaml [hunk 1]
```diff
@@ -260,6 +257,9 @@ spec:
             successThreshold: {{ .Values.envoy.readinessProbe.successThreshold }}
             failureThreshold: {{ .Values.envoy.readinessProbe.failureThreshold }}
           {{- end }}
+          {{- if .Values.envoy.customReadinessProbe }}
```
Comment (andresbono) [inline bitnami/contour/templates/envoy/deployment.yaml:260]: Same comment than in the template above:

```diff
diff --git a/bitnami/contour/templates/envoy/deployment.yaml b/bitnami/contour/templates/envoy/deployment.yaml
index f0e9ce465d..b0b349809f 100644
--- a/bitnami/contour/templates/envoy/deployment.yaml
+++ b/bitnami/contour/templates/envoy/deployment.yaml
@@ -246,7 +246,9 @@ spec:
               {{- end }}
               name: metrics
               protocol: TCP
-          {{- if .Values.envoy.readinessProbe.enabled }}
+          {{- if .Values.envoy.customReadinessProbe }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.envoy.customReadinessProbe "context" $) | nindent 12 }}
+          {{- else if .Values.envoy.readinessProbe.enabled }}
           readinessProbe:
             httpGet:
               path: /ready
```

**Review by andresbono — COMMENTED** [review]
Thank you for your contribution!! Could you please check my suggestions?

File: bitnami/contour/templates/envoy/daemonset.yaml [hunk 2]
```diff
@@ -248,6 +248,9 @@ spec:
             successThreshold: {{ .Values.envoy.readinessProbe.successThreshold }}
             failureThreshold: {{ .Values.envoy.readinessProbe.failureThreshold }}
           {{- end }}
+          {{- if .Values.envoy.customReadinessProbe }}
```
[the remark this answers is not in the document]
Comment (seth-epps) [reply bitnami/contour/templates/envoy/daemonset.yaml]: addressed in https://github.com/bitnami/charts/pull/24860/commits/a7b55bb9d4aa96e6afc1407a0eb81a5838a75853

File: bitnami/contour/templates/envoy/deployment.yaml (hunk 1 above)
Comment (seth-epps) [reply bitnami/contour/templates/envoy/deployment.yaml]: addressed in https://github.com/bitnami/charts/pull/24860/commits/a7b55bb9d4aa96e6afc1407a0eb81a5838a75853

**Comment by seth-epps** [comment]
with https://github.com/bitnami/charts/pull/24860/commits/50298dc942403856f006c68bb68820929e9835f0 running the template still behaves as expected with the custom probe block
#### DaemonSet  + No Custom Probe
```sh
❯ cat envoy-custom-readiness.yaml
envoy:
  kind: daemonset
  # custom envoy readiness check to hit a different port
  # customReadinessProbe:
  #   failureThreshold: 3
  #   httpGet:
  #     path: /ready
  #     port: 8003
  #     scheme: HTTP
```
```json
❯ helm template contour-envoy ../../../bitnami-charts/bitnami/contour -f envoy-custom-readiness.yaml | yq 'select(.kind == "DaemonSet" and .metadata.name == "contour-envoy-envoy") | .spec.template.spec.containers[] | .readinessProbe' -o json
{
  "exec": {
    "command": [
      "pgrep",
      "contour"
    ]
  },
  "initialDelaySeconds": 10,
  "periodSeconds": 3,
  "timeoutSeconds": 1,
  "successThreshold": 1,
  "failureThreshold": 3
}
{
  "httpGet": {
    "path": "/ready",
    "port": 8002
  },
  "initialDelaySeconds": 10,
  "periodSeconds": 3,
  "timeoutSeconds": 1,
  "successThreshold": 1,
  "failureThreshold": 3
}
```
#### DaemonSet + Custom Probe
```sh
❯ cat envoy-custom-readiness.yaml
envoy:
  kind: daemonset
  # custom envoy readiness check to hit a different port
  customReadinessProbe:
    failureThreshold: 3
    httpGet:
      path: /ready
      port: 8003
      scheme: HTTP
```
```json
❯ helm template contour-envoy ../../../bitnami-charts/bitnami/contour -f envoy-custom-readiness.yaml | yq 'select(.kind == "DaemonSet" and .metadata.name == "contour-envoy-envoy") | .spec.template.spec.containers[] | .readinessProbe' -o json
{
  "exec": {
    "command": [
      "pgrep",
      "contour"
    ]
  },
  "initialDelaySeconds": 10,
  "periodSeconds": 3,
  "timeoutSeconds": 1,
  "successThreshold": 1,
  "failureThreshold": 3
}
{
  "failureThreshold": 3,
  "httpGet": {
    "path": "/ready",
    "port": 8003,
    "scheme": "HTTP"
  }
}
```
#### Deployment + Custom Probe
```sh
❯ cat envoy-custom-readiness.yaml
envoy:
  kind: deployment
  # custom envoy readiness check to hit a different port
  customReadinessProbe:
    failureThreshold: 3
    httpGet:
      path: /ready
      port: 8003
      scheme: HTTP
```
```json
❯ helm template contour-envoy ../../../bitnami-charts/bitnami/contour -f envoy-custom-readiness.yaml | yq 'select(.kind == "Deployment" and .metadata.name == "contour-envoy-envoy") | .spec.template.spec.containers[] | .readinessProbe' -o json
{
  "exec": {
    "command": [
      "pgrep",
      "contour"
    ]
  },
  "initialDelaySeconds": 10,
  "periodSeconds": 3,
  "timeoutSeconds": 1,
  "successThreshold": 1,
  "failureThreshold": 3
}
{
  "failureThreshold": 3,
  "httpGet": {
    "path": "/ready",
    "port": 8003,
    "scheme": "HTTP"
  }
}
```

**Review by andresbono — APPROVED** [review]
Thank you for addressing the suggestions and fixing this bug! LGTM

**Pull request merged**